### ISSUE TITLE
Istio-1.0: Helm template creates a manifest that contains references to release 0.8

### ISSUE THREAD

**Issue by holgero** [istio/istio#6816]
**Describe the bug**
When I execute the first command line described in https://preliminary.istio.io/docs/setup/kubernetes/helm-install.html#option-1-install-with-helm-via-helm-template the generated manifest contains references to release 0.8

**Expected behavior**
The generated manifest should only contain references to release 1.0 (probably with some snapshot identifier).

**Steps to reproduce the bug**
1. Download http://bit.do/istio10-latest-daily and extract it.
1. install helm
1. run `helm template install/kubernetes/helm/istio --name istio --namespace istio-system > $HOME/istio.yaml`
1. Check the generated manifest file for references to 0.8: `grep '0\.8' $HOME/istio.yaml`

**Version**
Output of `istioctl version`:
```
Version: 1.0.0-snapshot.0
GitRevision: cf65e19c6a54df14d3a92de2221cc1e5153825de
User: root@5b39c9818787
Hub: docker.io/istio
GolangVersion: go1.10.1
BuildStatus: Clean
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-05-12T04:12:12Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
```
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

**Is Istio Auth enabled or not?**
Helm command line is as above: `helm template install/kubernetes/helm/istio --name istio --namespace istio-system > $HOME/istio.yaml`

**Environment**
Output of `uname -a` (slightly redacted):
```
Darwin hostname 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64
```
Didn't yet start to use a cloud vendor for this.

**Comment by holgero** [comment]
Seems this has been noticed on the mailing list, too: https://groups.google.com/d/msg/istio-dev/3i-xlzmjKmo/i6ogFI_tCAAJ
When I apply the workaround mentioned there, the docker image tags no longer contain the label with 0.8. But still grep finds occurrences of '0.8' in the manifest (all in the values for a property named 'chart'). I am not sure if this is important, or can/should be ignored.

**Comment by ymesika** [comment]
Resolved by #6867
